.github/workflows/build_and_test.yml

```
@@ -197,3 +197,29 @@ jobs:
         with:
           config: .github/.licenserc.yaml
 
+  macos-c++-check:
+    name: "C++ Test on macOS"
+    strategy:
+      fail-fast: false
+      matrix:
+        version: [12, 14]
+
+    runs-on: macos-${{ matrix.version }}
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Install dependencies
+      run: |
+        brew update
+        brew install protobuf
+
+    - name: Test
```

```suggestion
    name: "C++ Test on macOS"
    strategy:
      fail-fast: false
      matrix:
        version: [12, 14]
    runs-on: macos-${{ matrix.version }}
    steps:
    - name: Checkout repository
      uses: actions/checkout@v4
    - name: Install dependencies
      run: |
        brew update
        brew install protobuf
    - name: Test
```

Let's make it compact